### ISSUE TITLE
Union of ComplexRegion and Interval returns ComplexRegion

### ISSUE THREAD

**Issue by gxyd** [sympy/sympy#10419]
```
>>> c1 = ComplexRegion(Interval(0, 1)*Interval(0, 2*S.Pi), polar=True)
>>> c1.union(Interval(2, 4)) == c1
True
```

I think this should return a `Union` of `ComplexRegion` and the `Interval`

**Comment by gxyd** [comment]
@aktech . Was this done intentionally?

**Comment by aktech** [comment]
> @aktech . Was this done intentionally?

No.

**Comment by aktech** [comment]
It was due to this commit https://github.com/kevalds51/sympy/commit/6e002aaefd6d9efccf4cb0540911a2bdd43c92de

**Comment by gxyd** [comment]
This is the one i suggested :)

**Comment by aktech** [comment]
Oh, I see. Here is the fix #10420
